llmfoundry/callbacks/async_eval_callback.py

```
@@ -0,0 +1,274 @@
+            break
+
+    if not checkpointer:
+        log.warning('No checkpoint saver callback found')
+        return None
+
+    if event.name == Event.FIT_END:
+        # Use the latest symlink for the end of training
+        return checkpointer.latest_filename
+
+    if not checkpointer.saved_checkpoints:
+        log.warning('No saved checkpoints found on the checkpointer')
+        return None
+
+    return checkpointer.saved_checkpoints[-1]
+
+
+def get_eval_parameters(
+    parameters: Dict[str, Any],
+    checkpoint: str,
+    training_run_name: str,
+) -> Dict[str, Any]:
+    """Get the parameters needed for the eval run.
+
+    Args:
+        parameters: The parameters from the training run
+        checkpoint: The path to the latest checkpoint
+        training_run_name: The name of the training run
+
+    Returns:
+        The parameters needed for the eval run as a dict
+    """
+    looking_for = REQUIRED_PARAMS_FOR_EVAL.copy()
+
+    # Go through all parameters and pull out the ones needed for eval
+    subset_keys = {}
+    for key in parameters:
+        if key in OPTIONAL_PARAMS_FOR_EVAL:
+            subset_keys[key] = parameters[key]
+        elif key in REQUIRED_PARAMS_FOR_EVAL:
+            subset_keys[key] = parameters[key]
+            looking_for.remove(key)
+
+    if looking_for:
+        raise Exception(
+            f'Missing the following required parameters for async eval: {looking_for}\n{parameters}'
+        )
+
+    # Convert the save_folder to a load_path
+    subset_keys['load_path'] = checkpoint
+
+    for logger, config in subset_keys.get('loggers', {}).items():
+        if logger == 'wandb':
+            config['group'] = config.pop('name', training_run_name)
+
+        # mlflow currently does not support grouping, so this will just launch
+        # a new mlflow run
+
+    # Create new eval models list
+    model = subset_keys.pop('model')
+    new_models = {'model_name': model.get('name'), 'model': model}
```

is `model_name` optional? if so this is fine, otherwise it should raise a nice error if `model.get('name')` gives `None`

---

eek, yea this is required for eval